File: python/Venniyammal/Inheritance and classes/bike.py
from Vehicle import *
from  twowheeler import *
#class to define Bike functions
class bike(Vehicle,twowheeler):
    
    
    def bikeexecution():
        ch=0
        f=False
        speed = 10
        maxspeed=110
        while(ch !=4):
            try:
                print("Provide me an instruction to drive ")
                ch=int(input("\n\t\t 1.To start 2.To increase the  speed 3.To turn 4.To stop the bike "))
                if(ch==1 and f==False):
                    v=Vehicle("bike")
                    f=True
                elif(ch == 2 ):
                    speed=speed+20
                    if(speed<=110):
                         print("Now the bike is running at the speed of ",speed)
 
                    else:
                        print("Running at the Maximum speed" )
                        speed=speed-20
                        continue
                elif(ch==3 and speed<=maxspeed):
                    speed=speed-30
                    if(speed<10):
                        print ("Running at initial speed")
                    else:
                        v.turn("bike")
                        print("Now the Speed is ",speed )
                        speed=speed+10
                        print("turn completed now the bike is running at the speed",speed)
                elif(ch==4):
                    t=twowheeler
                    speed=0
                    t.stop()
                  
            except ValueError:
                print("\n\n***Enter the valid choice***")
                continue
    
        
    
   
   
        
            

File: java/Venniyammal/Workspace/Assesment2/src/com/treselle/Bike.java
package opt.com.treselle;
//venni

import java.util.Scanner;

class Bike extends sound implements vehicle ,twowheeler
    {
     static int speed= 10;
	 int maxSpeed = 80;
	 int limit=64;
	 // creating function for Increase and Decrease the Speed
	 
	 void increaseSpeed() 
	    { 
		    Exception e = new Exception();
	         speed = 10+speed;
		    /* while(limit!=speed)
		    {
		     System.out.println("\n \t Now the bike is running at the Speed of " + speed + " Km/h.");
			}*/
		System.out.println("\n \t Now the bike is running at the Speed of " + speed + " Km/h.");
		  if(speed==80)
		    {
			 System.out.println(" Alert !!! Alert!!! Your Exceeding the speed limit");
			}
		   else if ( speed>=64)
			{
			 System.out.println("\n\t Your running at the 80% of the speed");
			 }
		    
	    }
	void decreaseSpeed()
	    {
	         speed = speed-20;
		     System.out.println("\n\t Now the Bike is running at the Speed of " + speed +"  Km/h." );
		}
		
		
	 
	 // implementation of Interface methods 
	 
	
		
	 public void start()
	     {
		     releaseStand();
			 
		     System.out.println("\n \t Now the bike Started at the Initial Speed of 10 km/h");
		 }
	 public void turn()
	     { 
		    
		   decreaseSpeed();
		   System.out.println( " \n \t Turn Completed , and Increasing the Speed");
		   increaseSpeed();
		  }
	 public void stop()
		 {
		    speed =10;
		    System.out.println("\n \t\t now the speed decreased to 10 Km/h.");
			releaseStand();
			putStand();
		    System.out.println("\n \t Now the bike is stopped ");
			 }
	public void releaseStand()
			{
			System.out.println("\n \t Release the side Stand ");
			}
	public void putStand()
			{
			System.out.println("\n \t Lock the side Stand ");
			}
	void horn()
        {
         System.out.println("\n \t Beeeeeeeep");
        }
		
		 Scanner in = new Scanner(System.in);
		public void excecute()
		{
		int n =1;
     	while(n!=5)
		{
		   
			System.out.println("\n  *!*!*!*! provide me an instuction to drive the Bike. !*!*!*!*!*");
			System.out.println("\n \n \t \tPress 1 : To Start \n\t\tPress 2 : To increase the speed \n\t\tPress 3 : To return \n\t\tPress 4 : To Stop \n\t\tPress 5 : To Halt");
			
            int ch = in.nextInt();
			switch(ch)
			{
			case 1:
				start();
				break;
			case 2:
			    increaseSpeed();
				break;
			case 3:
				turn();
				break;
			case 4:
				stop();
				break;
			}
		n=ch;
		
		}
     }
		 
	}
	